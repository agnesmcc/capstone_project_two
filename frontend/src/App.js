import { React, useEffect } from "react";
import { Routes, Route } from 'react-router-dom';
import './App.css';
import useLocalStorage from "./useLocalStorage";
import Api from "./Api";
import Home from './Home';
import LoginPage from "./LoginPage";
import SignupPage from "./SignupPage";

function App() {
  const [token, setToken] = useLocalStorage('token', null);
  
  useEffect(() => {
    if (token) {
      Api.token = token;
    }
  }, [token]);

  return (
    <div className="App">
        <Routes>
          <Route exact path="/" element={<Home />} />
          <Route path="/login" element={<LoginPage setToken={setToken}/>} />
          <Route path="/signup" element={<SignupPage setToken={setToken}/>} />
        </Routes>
    </div>
  );
}

export default App;
