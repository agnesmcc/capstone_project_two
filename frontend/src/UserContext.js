import { createContext } from "react";

export const UserContext = createContext();

/**
 * The UserProvider component is responsible for providing the user information to the
 * components that request it. It expects a children prop which is the JSX that will be
 * wrapped by the UserProvider component, and a user prop which is an object with all the
 * user information.
 *
 * @param {Object} children the JSX that will be wrapped
 * @param {String} token the user token
 * @param {Object} user the user object
 * @returns {React.Component} The UserProvider component
 */
export const UserProvider = ({ children, token, user}) => {
  return (
    <UserContext.Provider token={{token}} value={{ user }}>
      {children}
    </UserContext.Provider>
  );
};
