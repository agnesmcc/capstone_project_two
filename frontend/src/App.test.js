import { render, screen } from '@testing-library/react';
import App from './App';
import { MemoryRouter } from 'react-router-dom';
import { UserProvider } from './UserContext';
import { QueryClient, QueryClientProvider } from 'react-query';

const queryClient = new QueryClient();

const { container } = render(
  <MemoryRouter initialEntries={['/']}>
    <UserProvider>
    <QueryClientProvider client={queryClient}>
      <App />
    </QueryClientProvider>
    </UserProvider>
  </MemoryRouter>
);

test('renders App using MemoryRouter', () => {  
  expect(container).toMatchSnapshot();
});

// test('clicking on Signup navigates to the signup page', () => {
//   const signupLink = screen.getByText('Signup');
//   signupLink.click();
//   expect(window.location.pathname).toBe('/signup');
// });
