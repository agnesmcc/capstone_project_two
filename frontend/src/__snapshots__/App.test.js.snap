// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App using MemoryRouter 1`] = `
<div>
  <div>
    <div>
      <nav
        class="navbar"
      >
        <div
          class="container-fluid"
        >
          <a
            class="navbar-brand"
            href="/"
          >
            eBid
          </a>
          <ul
            class="nav"
          >
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/login"
              >
                Login
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                class="nav-link"
                href="/signup"
              >
                Signup
              </a>
            </li>
          </ul>
        </div>
      </nav>
    </div>
    <div>
      <div>
        <div
          class="listings"
        />
      </div>
    </div>
  </div>
</div>
`;
